source/agora/node/Registry.d

```diff
@@ -834,31 +887,159 @@ private struct ZoneData
             "FOREIGN KEY(pubkey) REFERENCES registry_%s_utxo(pubkey) ON DELETE CASCADE, " ~
             "PRIMARY KEY(pubkey, address))", zone_name, zone_name);
 
+        string query_prev_type = format("SELECT * FROM registry_%s_utxo " ~
+            "WHERE utxo = ?", zone_name);
+
+        bool was_primary;
+
+        try
+            was_primary = this.db.execute(query_prev_type, Hash.init).empty;
+        catch (Exception)
+            was_primary = false;
+
+        if (this.type == ZoneType.primary)
+        {
+            if (!was_primary)
+            {
+                this.db.execute(
+                    format("DROP TABLE IF EXISTS registry_%s_addresses", zone_name));
+                this.db.execute(
+                    format("DROP TABLE IF EXISTS registry_%s_utxo", zone_name));
+            }
+
+            this.config.fromConfig(this.root);
+            this.updateSOA();
+        }
+        else if (this.type == ZoneType.secondary)
+        {
+            if (was_primary)
+            {
+                this.db.execute(
+                    format("DROP TABLE IF EXISTS registry_%s_utxo", zone_name));
+                this.db.execute(
+                    format("DROP TABLE IF EXISTS registry_%s_addresses", zone_name));
+            }
+
+            // DNS resolver is used to get SOA RR and performing AXFR
+            auto peer_addrs = this.config.primary_servers.map!(
+                peer => Address("dns://" ~ peer)
+            ).array();
+            this.resolver = node.makeDNSResolver(peer_addrs);
```

Comment:
Why doesn't the `Node` do that itself ?

Reply:
It's per-Zone. Each zone can have different list of primaries.